configure.ac

```
@@ -3747,6 +3747,18 @@ AS_IF([test x"$MJIT_SUPPORT" = "xyes"],
 
 AC_SUBST(MJIT_SUPPORT)
 
+AC_ARG_ENABLE(compile-commands,
+        AS_HELP_STRING([--enable-compile-commands],
+        [enable compile commands for language servers (only works with clang)[default=no]]),
+        [COMPILE_COMMANDS=$enableval], [COMPILE_COMMANDS=no])
+
+AS_IF([test x"$COMPILE_COMMANDS" = "xyes" -a x"$CC" != "xclang"],
+      [AC_MSG_ERROR([compile commands only works with clang])],
+      [])
+
+AS_IF([test x"$COMPILE_COMMANDS" = "xyes"],
+	[RUBY_APPEND_OPTIONS(CPPFLAGS, "-MJ \$@.json")])
```

This does not allow to use clang-14, etc.
The most reliable way to check if a compiler supports an option is to ask the compiler.

```suggestion
AS_IF([test x"$COMPILE_COMMANDS" = "xyes"], [
    COMPILE_COMMANDS=no
    RUBY_TRY_CFLAGS([-MJ conftest.o.json], [
        AS_IF([test -f conftest.o.json], [
            rm -f conftest.o.json
            COMPILE_COMMANDS=yes
        ])
    ])
    AS_IF([test x"$COMPILE_COMMANDS" = "xno"],
        [AC_MSG_ERROR([compile commands does not work with this compiler])])
    RUBY_APPEND_OPTIONS(XCFLAGS, '-MJ $@.json')
])
```